fwoa-tools/src/migrationVerify.ts

```diff
@@ -91,26 +87,70 @@ async function verifyFolderImport(): Promise<void> {
       }
 
       // Each resource file can contain a number of resource objects
-      const allResources: string[] = resourceFile.Body!.toString().split('\n');
+      const allResources: string[] = resourceFile.Body!.toString().trimEnd().split('\n');
       for (let j = 0; j < allResources.length; j += 1) {
         // eslint-disable-next-line security/detect-object-injection
         const resource = JSON.parse(allResources[j]);
-        const id = resource.id;
-        const resourceInHL = await healthLakeClient.get(
-          `${DATASTORE_ENDPOINT}/${resource.resourceType}/${id}`
-        );
+        comparisonQueue.push(`${resource.resourceType}/${resource.id}`);
         logs.write(
-          `\n${new Date().toISOString()}: Retrieved resource at ${resourcePath} line ${j} from datastore, comparing to FWoA...`
+          `\n${new Date().toISOString()}: Retrieved resource at ${resourcePath} line ${j} from datastore, queuing for comparison...`
         );
-        if (!(await verifyResource(fhirClient, resourceInHL.data, id, resource.resourceType))) {
-          throw new Error(`Resources in FWoA and AHL do not match, ${resourcePath}`);
+        if (comparisonQueue.length >= HEALTHLAKE_BUNDLE_LIMIT) {
+          await compareResources(comparisonQueue);
+          comparisonQueue = [];
         }
       }
     }
+    if (comparisonQueue.length >= HEALTHLAKE_BUNDLE_LIMIT) {
```

Comment:
Should this be `comparisonQueue.length > 0`?

Reply:
Yup, it should be, thanks for catching that.